### PR TITLE
BKNDLSS-31849 zooming in/out pdf files 

### PR DESCRIPTION
**Pull request by StasivPavlo**
image

### CODE REVIEW

File: components/bl-pdfViewer-component/src/icons.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    </svg>
+  );
+}
+
+export function MinusIcon() {
+  return (
+    <svg className="button-icon" viewBox="0 0 20 20" fill="none">
+      <g id="SVGRepo_bgCarrier"></g>
+      <g id="SVGRepo_tracerCarrier"></g>
+      <g id="SVGRepo_iconCarrier">
```
Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/icons.js:86]: Do we need these ids? `SVGRepo...`

File: components/bl-pdfViewer-component/src/icons.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        <path fill="#ffffff" d="M18 10a1 1 0 01-1 1H3a1 1 0 110-2h14a1 1 0 011 1z"></path>
```
Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/icons.js:87]: Some user in future ask to change colors of these icons. Is it possible now? If no could we add it in easy way(without properties)?

File: components/bl-pdfViewer-component/src/subcomponents.js [hunk 3]
```diff
@@ -57,62 +67,108 @@ export function Controls(props) {
...
   );
 }
 
-function NextButtonIcon() {
+function ScaleControls({ scale, setScale }) {
+  const onMinusScaleClick = () => {
+    if (scale > 0.1) {
+      setScale(state => Number((state - 0.1).toFixed(1)));
+    }
+  };
```
Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/subcomponents.js:104]: enpty line

Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/subcomponents.js:104]:
```suggestion
  const onMinusScaleClick = () => setScale(s => +Math.max(0.1, s - 0.1).toFixed(1))
```

File: components/bl-pdfViewer-component/src/subcomponents.js [hunk 4]
```diff
@@ -57,62 +67,108 @@ export function Controls(props) {
...
 
-function PrevButtonIcon() {
+function OtherControls({ pdfUrl, showDownloadButton, showPrintButton }) {
+  const printButtonClick = () => {
+    printJS({
+      printable         : pdfUrl,
+      type              : 'pdf',
+      onPrintDialogClose: () => document.getElementById('printJS').remove(),
+    });
+  };
```
Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/subcomponents.js:131]:
```suggestion
  const printButtonClick = () =>  printJS({
      printable         : pdfUrl,
      type              : 'pdf',
      onPrintDialogClose: () => document.getElementById('printJS').remove(),
    });

```

File: components/bl-pdfViewer-component/src/subcomponents.js [hunk 5]
```diff
@@ -57,62 +67,108 @@ export function Controls(props) {
...
-      aria-hidden="true"
-      viewBox="0 0 24 24">
-      <path d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"></path>
-    </svg>
+    <div className="controls__other">
+      { showDownloadButton && (
+        <a
+          className="controls__button"
+          href={ pdfUrl }
+          download="pdf">
```
Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/subcomponents.js:139]: one line?

File: components/bl-pdfViewer-component/src/subcomponents.js [hunk 6]
```diff
@@ -57,62 +67,108 @@ export function Controls(props) {
...
+          <DownloadIcon/>
+        </a>
+      ) }
+      { showPrintButton && (
+        <button
+          className="controls__button"
+          onClick={ printButtonClick }>
```
Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/subcomponents.js:146]: one line?

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-pdfViewer-component/src/subcomponents.js [hunk 7]
```diff
@@ -57,62 +67,94 @@ export function Controls(props) {
...
+        onBlur={ onSubmit }
+      />
+      <span>/{ pageCount }</span>
+    </div>
+  );
+}
+
+function ScaleControls({ scale, setScale }) {
+  const onMinusScaleClick = () => setScale(s => +Math.max(0.1, s - 0.1).toFixed(1));
+
```
Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/subcomponents.js:101]:
```suggestion
```

File: components/bl-pdfViewer-component/src/subcomponents.js [hunk 8]
```diff
@@ -57,62 +67,93 @@ export function Controls(props) {
...
+      <span>{ scale }</span>
+      <button className="controls__button" onClick={ onPlusScaleClick } disabled={ scale >= 10 }>
+        <PlusIcon/>
+      </button>
+    </div>
+  );
+}
+
+function OtherControls({ pdfUrl, showDownloadButton, showPrintButton }) {
+  const printButtonClick = () =>  printJS({
```
Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/subcomponents.js:117]:
```suggestion
  const printButtonClick = () => printJS({
```

File: components/bl-pdfViewer-component/src/subcomponents.js [hunk 9]
```diff
@@ -57,62 +67,93 @@ export function Controls(props) {
...
+      </button>
+    </div>
+  );
+}
+
+function OtherControls({ pdfUrl, showDownloadButton, showPrintButton }) {
+  const printButtonClick = () => printJS({
+    printable         : pdfUrl,
+    type              : 'pdf',
+    onPrintDialogClose: () => document.getElementById('printJS').remove(),
```
Comment (v-excelsior) [inline components/bl-pdfViewer-component/src/subcomponents.js:120]: Is it works correct if we have multiple pdf files on the screen?

Comment (StasivPavlo) [reply components/bl-pdfViewer-component/src/subcomponents.js:120]: Yes. This element is mounted when the print dialog is opened, and at that moment there is no way to open this window for another component. So when this event fires, there can only be one element

**Pull request merged**